src/client/components/app/LineDetailsDisplay/PositionsTable.tsx

```diff
@@ -232,7 +242,127 @@ export const PositionsTable = ({ positions, displayLine = false }: PositionsProp
+  // Returns a list of transactions to display on positions table for proposals
+  const getUserProposalActions = (proposal: CreditProposal) => {
+    const approveMutualConsent = {
+      name: t('components.transaction.add-credit.accept-terms'),
+      handler: depositHandler,
+      disabled: false,
+    };
+
+    const revokeMutualConsent = {
+      name: t('components.transaction.revoke-consent.cta'),
+      handler: revokeConsentHandler,
+      disabled: false,
+    };
+
+    // Display button to approve proposal for the borrower of the proposal
+    if (userRoleMetadata.role === BORROWER_POSITION_ROLE) {
+      return [approveMutualConsent];
+    }
+
+    // Display button to cancel proposal for the maker of the proposal
+    if (getAddress(proposal.maker) === userWallet) {
+      return [revokeMutualConsent];
+    }
+
+    return [];
+  };
+
+  const formattedPositionsAndProposals = _.flatten(
+    positions?.map((position) => {
+      // get position (not in PROPOSED_STATUS)
+      const positionToDisplay = {
+        deposit: humanize('amount', position.deposit, position.token.decimals, 2),
+        drate: `${normalizeAmount(position.dRate, 2)} %`,
+        frate: `${normalizeAmount(position.fRate, 2)} %`,
+        line: (
+          <RouterLink to={`/${currentNetwork}/lines/${position.line}`} key={position.line} selected={false}>
+            {formatAddress(position.line)}
+            <RedirectLinkIcon />
+          </RouterLink>
+        ),
+        status: position.status,
+        principal: humanize('amount', position.principal, position.token.decimals, 2),
+        interest: humanize('amount', position.interestAccrued, position.token.decimals, 2),
+        lender: (
+          <RouterLink to={`/${currentNetwork}/portfolio/${position.lender}`} key={position.id} selected={false}>
+            {formatAddress(getENS(position.lender, ensMap)!)}
+            <RedirectLinkIcon />
+          </RouterLink>
+        ),
+        token: (
+          <RouterLink
+            key={position.token.address}
+            to={`https://etherscan.io/address/${position.token.address}`}
+            selected={false}
+          >
+            {position.token.symbol}
+          </RouterLink>
+        ),
+        actions: (
+          <ActionButtons
+            value1={position.id}
+            actions={position.status !== PROPOSED_STATUS ? getUserPositionActions(position) : []}
+          />
+        ),
+      };
+      // generate list of proposals from each position in PROPOSED_STATUS
+      const proposalsToDisplay =
+        position.status === PROPOSED_STATUS
+          ? Object.values(position.proposalsMap)
+              .filter((proposal) => proposal.revokedAt === null)
+              .map((proposal) => {
+                return {
+                  deposit: humanize('amount', proposal.args[2], position.token.decimals, 2),
+                  drate: `${normalizeAmount(proposal.args[0], 2)} %`,
```

Comment:
can u add inline comments on whatthe `args[m]` are. A top level comment saying what `propsal.args` is - an ordered list of tx inputs for the proposal. And then a comment on each line used saying what the var is.

Or destructure the array into named variables before using.

Reply:
Destructed the array in named variables. 

I'd already done this in `RevokeConsentTx` component and I think it's cleaner, self-documenting code.